### PR TITLE
Support metric

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**

### CODE REVIEW

**Review by breezewish — APPROVED** [review]
The rest LGTM

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+func (m *Metrics) Unregister() {
+	prometheus.Unregister(m.TableNumGauge)
+	prometheus.Unregister(m.SnapshotTotalSizeCounter)
+	prometheus.Unregister(m.SnapshotLoadedSizeCounter)
+	prometheus.Unregister(m.IncrementPeddingSizeGauge)
+	prometheus.Unregister(m.IncrementLoadedCounter)
+	prometheus.Unregister(m.TableVersionsCounter)
+	prometheus.Unregister(m.ErrorCounter)
+}
```
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply pkg/metrics/metrics.go]: https://github.com/pingcap-inc/tidb2dw/blob/12c51bc2f3bd3c7e16401154f22e2962a7567375/pkg/apiservice/server.go#L60 unregister when server receive quit signal, not quite sure whether it is necessary

File: pkg/metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+			Name:      "table_num",
+			Help:      "number of replication tables",
+		})
+	m.SnapshotTotalSizeCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: Namespace,
+			Name:      "snapshot_total_size",
+			Help:      "total file size of snapshot",
+		}, []string{"table"})
+	m.SnapshotLoadedSizeCounter = prometheus.NewCounterVec(
```
[the remark this answers is not in the document]
Comment (Lloyd-Pottiger) [reply pkg/metrics/metrics.go]: We need to read file to get rows, which maybe a heavy operator.

File: pkg/metrics/metrics.go [hunk 3]
```diff
@@ -11,85 +11,79 @@ const (
...
 
-func (m *Metrics) Register() {
-	prometheus.MustRegister(m.TableNumGauge)
-	prometheus.MustRegister(m.SnapshotTotalSizeCounter)
-	prometheus.MustRegister(m.SnapshotLoadedSizeCounter)
-	prometheus.MustRegister(m.IncrementPeddingSizeGauge)
-	prometheus.MustRegister(m.IncrementLoadedCounter)
-	prometheus.MustRegister(m.TableVersionsCounter)
-	prometheus.MustRegister(m.ErrorCounter)
+func Register() {
```
Comment (breezewish) [inline pkg/metrics/metrics.go:69]: `InitMetrics` and `Register` to `func init()`. Then these metrics never needs unregister :)

**Pull request merged**